base/arrayshow.jl

```diff
@@ -299,14 +299,12 @@ end
 
 # print_array: main helper functions for show(io, text/plain, array)
 # typeinfo agnostic
-
-# 0-dimensional arrays
-print_array(io::IO, X::AbstractArray{T,0} where T) =
-    isassigned(X) ? show(io, X[]) :
-                    print(io, undef_ref_str)
-
+# Note that this is for showing the content inside the array, and for `MIME"text/plain".
+# There are `show(::IO, ::A) where A<:AbstractArray` methods that don't use this
+# e.g. show_vector, show_zero_di, show_zero_dim
+print_array(io::IO, X::AbstractArray{<:Any, 0}) =
+    isassigned(X) ? show(io, X[]) : print(io, undef_ref_str)
```

Comment:
No code here changed - I just added a comment, becuase I was confused by the role this function was playing (I tried making the change here and it led to weird behaviour